evap/staff/tools.py

```diff
@@ -195,6 +195,9 @@ def bulk_update_users(request, user_file_content, test_run):  # noqa: PLR0912
                 user, deletable_users + users_to_mark_inactive, test_run
             ):
                 messages.warning(request, message)
+        for user in users_to_mark_inactive:
+            for message in remove_inactive_participations(user, test_run):
+                messages.warning(request, message)
```

Comment:
Ah, this is a bit tricky: The bulk update is supposed to process hundreds or thousands of users. This code now breaks it down to "a thousand times, process one user", which is fine in isolation, but the logic below for "process one user" (in `remove_inactive_participations`) happens to make very latency-heavy database queries (`user.evaluations_participating_in.filter` boils down to a sql join query).

We might still be fine here for now if we just add a mitigation later (i.e., rewrite the code to do one big query instead of many small ones). I'm not sure about current performance and performance requirements.

@janno42 @niklasmohrin @Kakadus thoughts? Might be fine for now. User Bulk Update isn't that often -- but you know me :D We can create a follow up if you're fine with it.

Reply:
The same problem already exists in the loop right before right? In that case, I am fine with (maybr even prefer) leaving it as is for now